runtime/bcutil/cfreader.c

```diff
@@ -418,33 +418,51 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
 			}
 
 			annotations = (J9CfrAttributeRuntimeVisibleAnnotations *)attrib;
+			annotations->numberOfAnnotations = 0;
+			annotations->annotations = NULL;
 			annotations->rawAttributeData = NULL;
 			annotations->rawDataLength = 0; /* 0 indicates attribute is well-formed */
-			CHECK_EOF(2);
-			NEXT_U16(annotations->numberOfAnnotations, index);
 
-			if (!ALLOC_ARRAY(annotations->annotations, annotations->numberOfAnnotations, J9CfrAnnotation))
-			{
-				return BCT_ERR_OUT_OF_ROM;
-			}
+			/* In the case of a malformed attribute numberOfAnnotations may not exist even if the file did not end. 
+			 * Don't read a bogus numberOfAnnotations.
+			 */
+			if (length > 1) {
```

Comment:
Since a length of 1 is not long enough to have a valid `numberOfAnnotations` I wanted to skip the sections where `numberOfAnnotations` is set. Since this 1 byte is not being read the annotation will enter the error handling code since `(index != end)`.
I will add some additional comments to make this clear.